src/core/Akka.Remote/Endpoint.cs

```diff
@@ -1141,15 +1143,13 @@ private void Initializing()
             {
                 if (failure.Cause is InvalidAssociationException)
                 {
-                    PublishAndThrow(new InvalidAssociation(LocalAddress, RemoteAddress, failure.Cause),
-                        LogLevel.WarningLevel);
-                }
-                else
-                {
-                    PublishAndThrow(
-                        new EndpointAssociationException(string.Format("Association failed with {0}",
-                            RemoteAddress)), LogLevel.DebugLevel);
+                    if (failure.Cause.InnerException == null)
```

Comment:
In here there is a mapping from InvalidAssociationException to **InvalidAssociation** exception. This mapping was here before! Again - no inner exception - situation is under control - we know what is happening.

Everything else mapped and propagated with the original exception inside. Btw old implementation was hiding it.